pkg/types/quickfiltertypes/filter.go

```diff
@@ -0,0 +1,146 @@
+package quickfiltertypes
+
+import (
+	"encoding/json"
+	"fmt"
+	v3 "github.com/SigNoz/signoz/pkg/query-service/model/v3"
+	"github.com/SigNoz/signoz/pkg/types"
+	"github.com/SigNoz/signoz/pkg/valuer"
+	"github.com/uptrace/bun"
+)
+
+type Signal struct {
+	valuer.String
+}
+
+var (
+	SignalTraces = Signal{valuer.NewString("TRACES")}
+	SignalLogs   = Signal{valuer.NewString("LOGS")}
+	//SignalMetrics     = Signal{valuer.NewString("METRICS")}
+	//SignalInfra       = Signal{valuer.NewString("INFRA")}
+	SignalApiMonitoring = Signal{valuer.NewString("API_MONITORING")}
+)
+
+// ValidSignals is a map of valid signal values
+var ValidSignals = map[string]bool{
+	SignalTraces.StringValue(): true,
+	SignalLogs.StringValue():   true,
+	//SignalMetrics.StringValue():       true,
+	//SignalInfra.StringValue():         true,
+	SignalApiMonitoring.StringValue(): true,
+}
+
+// IsValidSignal checks if a signal string is valid
+func IsValidSignal(signal string) bool {
+	return ValidSignals[signal]
+}
+
+// SignalFromString creates a Signal from a string
+func SignalFromString(s string) Signal {
+	return Signal{valuer.NewString(s)}
+}
+
+type StorableQuickFilter struct {
+	bun.BaseModel `bun:"table:quick_filter"`
+	types.Identifiable
+	OrgID  valuer.UUID `bun:"org_id,type:text,notnull"` // Changed from valuer.String to string
+	Filter string      `bun:"filter,type:text,notnull"` // Changed from valuer.String to string
+	Signal string      `bun:"signal,type:text,notnull"` // Changed from Signal to string
```

Comment:
```suggestion
	Signal Signal      `bun:"signal,type:text,notnull"` // Changed from Signal to string
```

The valuer has all methods defined to make it work as expected with databases and incoming http requests and whatnot